diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3f316d6..a45f44a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -2,12 +2,6 @@ cmake_minimum_required(VERSION 2.8)
 
 project(fifo_map LANGUAGES CXX)
 
-add_executable(unit
-    src/fifo_map.hpp test/thirdparty/catch/catch.hpp test/unit.cpp
-)
-
-target_include_directories(unit PRIVATE "test" "src" "test/thirdparty")
-
 if(MSVC)
     set(CMAKE_CXX_FLAGS
         "/EHsc"
